WordPress/Classes/ViewRelated/Blog/Blog Dashboard/ViewModel/BlogDashboardViewModel.swift

```diff
@@ -49,7 +49,7 @@ class BlogDashboardViewModel {
             return nil
         }
 
-        return DashboardDataSource(collectionView: viewController.collectionView) { [unowned self] collectionView, indexPath, item in
+        return DashboardDataSource(collectionView: viewController.collectionView) { [unowned self, unowned viewController] collectionView, indexPath, item in
```

Comment:
Even though `viewController` is a weak property, is it a strong reference here due to the local variable? I need to brush up on this!

Reply:
I admit, these ones were the hardest for me to understand too. 

This is the cycle we get without this fix:
<img width="1397" alt="image" src="https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/48cc13d3-d3c8-4944-a843-98bc2a5d6538">

Yes, `BlogDashboardViewModel` is holding a weak reference to `BlogDashboardViewController`:

```
final class BlogDashboardViewModel {
    private weak var viewController: BlogDashboardViewController?
```

However, when we capture  `BlogDashboardViewController` in a  `DashboardDataSource` closure, we capture it strongly. It creates a retain cycle of:
`BlogDashboardViewModel` -> `DashboardDataSource` -> `closure` -> `BlogDashboardViewController` -> `BlogDashboardViewModel`

Here's an isolated example you could run on playground:

```
class DataSource {
    let closure: () -> ()
    
    init(closure: @escaping () -> ()) {
        self.closure = closure
    }
}

class ViewModel {
    weak var viewController: ViewController?
    
    lazy var dataSource: DataSource? = {
        guard let viewController = viewController else {
            return nil
        }
        
        return DataSource {
            print(viewController)
        }
    }()
    
    deinit {
        print("ViewModel deinit")
    }
}

class ViewController {
    let viewModel: ViewModel
    
    init(viewModel: ViewModel) {
        self.viewModel = viewModel
        self.viewModel.viewController = self
        self.viewModel.dataSource?.closure()
    }
    
    deinit {
        print("ViewController deinit")
    }
}

ViewController(viewModel: ViewModel())
```

`ViewController` and `ViewModel` do not print deinit message until we do:

```
        return DataSource { [unowned viewController] in
            print(viewController)
        }
```

Reply:
Makes sense now. In the sample code above the retain cycle is due to the strong references:
`ViewController` > `ViewModel` > `DataSource` > `closure` > `ViewController`
So it hasn't got anything to do with the local variable.